homeassistant/components/reolink/__init__.py

```diff
@@ -192,6 +200,21 @@ async def async_check_firmware_update() -> None:
 
     hass.http.register_view(PlaybackProxyView(hass))
 
+    async def refresh(*args: Any) -> None:
+        """Request refresh of coordinator."""
+        await device_coordinator.async_request_refresh()
+
+    def async_privacy_mode_change() -> None:
+        """Request update when privacy mode is turned off."""
+        if host.privacy_mode and not host.api.baichuan.privacy_mode():
+            # The privacy mode just turned off, give the API 2 seconds to start
+            config_entry.async_on_unload(async_call_later(hass, 2, refresh))
```

Comment:
This is going to leak because nothing removes them when they are done until its unloaded

Reply:
I dont quite understand.
If the async_call_later fires after 2 minutes it will then clean itself after beeing done processing right?

Or do you mean each time this is run `config_entry.async_on_unload` will keep a refrence which will not be cleaned up?

Reply:
Would it be safe to just remove the `config_entry.async_on_unload` and only use the `async_call_later`?

Reply:
> Or do you mean each time this is run `config_entry.async_on_unload` will keep a refrence which will not be cleaned up?

Yes. The callable returned by `async_call_later` will be stored until the config entry unloads

Reply:
Thanks for pointing this out, did not see that.
I should have it fixed now with this commit: https://github.com/home-assistant/core/pull/135947/commits/041d689d51012b3ea5bac28c5cb0f2647e7476e2